BenchmarkVCFs/CompareBenchmarks.wdl

```diff
@@ -0,0 +1,428 @@
+    required_named = parser.add_argument_group('Required named arguments')
+    required_named.add_argument('--sample-ids', required=True, type=str, nargs='+')
+    required_named.add_argument('--configurations', required=True, type=str, nargs='+')
+    required_named.add_argument('--summaries', required=True, type=str, nargs='+')
+    args = parser.parse_args()
+    main(args.sample_ids, args.configurations, args.summaries, args.order_of_samples, args.order_of_configurations)
+EOF
+        python script.py --sample-ids ~{sep=' ' sample_ids} --configurations ~{sep=' ' configurations} --summaries ~{sep=' ' benchmark_summaries} ~{order_of_samples_arg} ~{sep=' ' order_of_samples_or_empty} ~{order_of_configurations_arg} ~{sep=' ' order_of_configurations_or_empty}
+    >>>
+
+    runtime {
+        docker: "us.gcr.io/broad-dsde-methods/benchmark_vcfs/compare_benchmarks:1.1.0"
+        preemptible: preemptible
+        memory: mem_gb + " GB"
+        disks: "local-disk 20 HDD"
+    }
+
+    output {
+        Array[File] gc_plots = glob("gc_plot_*.png")
+    }
+}
+
+
+
+task CompareBenchmarksTask {
+    input {
+        Array[String] sample_ids
+        Array[String] configurations
+        Array[File] benchmark_summaries
+        Array[String]? stratifiers
+
+        Boolean include_counts
+
+        Array[String]? order_of_samples
+        Array[String]? order_of_configurations
+        Array[Int]? deltas
+
+        Int mem_gb = 4
+        Int preemptible = 0
+    }
+
+    String stratifiers_arg = if !defined(stratifiers) then "" else "--stratifiers"
+    Array[String] stratifiers_or_empty = select_first([stratifiers, []])
+
+    String order_of_samples_arg = if !defined(order_of_samples) then "" else "--order-of-samples"
+    Array[String] order_of_samples_or_empty = select_first([order_of_samples, []])
+    String order_of_configurations_arg = if !defined(order_of_configurations) then "" else "--order-of-configurations"
+    Array[String] order_of_configurations_or_empty = select_first([order_of_configurations, []])
+
+    String deltas_arg = if !defined(deltas) then "" else "--deltas"
+    Array[Int] deltas_or_empty = select_first([deltas, []])
+
+    command <<<
+        set -xeuo pipefail
+        
+        source activate compare_benchmarks
+        
+        cat <<'EOF' > script.py
+import argparse
+import numpy as np
+import pandas as pd
+
+class ChainableOutput:
+    def __init__(self, file, sep:str):
+        self.file = file
+        self.separator = sep
+
+    def cells(self, cells:list):
+        self.file.write(self.separator.join([str(cell) for cell in cells]))
+        return self
+    def sep(self):
+        self.file.write(self.separator)
+        return self
+    def new_line(self):
+        self.file.write('\n')
+        return self
+
+def write_header(output:ChainableOutput, unique_sample_ids, unique_configurations, deltas, include_counts):
```

Comment:
so, I think this is all to get your nicely formatted table, and that's why this isn't just a simple pandas write?  A little ascii art comment might be helpful to clarify what the result of all this will be.

Reply:
Done